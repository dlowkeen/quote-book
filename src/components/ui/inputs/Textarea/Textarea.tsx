import React from 'react';

import * as styles from '../inputs.css';

interface ITextAreaProps {
  label: string;
  name: string;
  handleChange: (e: any) => void;
}

const Textarea: React.FunctionComponent<ITextAreaProps> = ({
  label,
  name,
  handleChange,
}) => (
  <div className={`${styles.formElement} ${styles.formElementText}`}>
    <label>{label}</label>
    <textarea name={name} onChange={handleChange} />
  </div>
);

export default Textarea;
