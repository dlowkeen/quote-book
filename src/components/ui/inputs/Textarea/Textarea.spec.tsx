import React from 'react';
import renderer from 'react-test-renderer';

import Textarea from './Textarea';

describe('The Textarea component', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <Textarea
          label='i have something i need to tell you'
          name='george'
          handleChange={() => 'handled'}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
