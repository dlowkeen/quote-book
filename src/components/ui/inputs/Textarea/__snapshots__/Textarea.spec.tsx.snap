// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Textarea component renders as expected 1`] = `
<div
  className="inputs__form-element__VT4Ts inputs__form-element-text__13VAX"
>
  <label>
    i have something i need to tell you
  </label>
  <textarea
    name="george"
    onChange={[Function]}
  />
</div>
`;
